src/xdsl/ir.py

```diff
@@ -18,7 +20,7 @@ class MLContext:
     _registeredAttrs: Dict[str, typing.Type[Attribute]] = field(
         default_factory=dict)
 
-    def register_op(self, op: typing.Type[Operation]) -> None:
+    def register_op(self, op: OperationType) -> None:
```

Comment:
Here, shouldn't this be `typing.Type[OperationType]`?
I'm actually not sure why this change would work.

Reply:
I think that both syntaxes work just fine. It might be that the newer syntax has been introduced in some recent Python versions.

Reply:
So I'm not understanding something.
Here, from what I understand, `op` will be an object of type `Operation` (or any superclass).
However, we want `op` to be a type that is a subclass of `Operation`.
Am I understand this right?

Reply:
Sory, you are absolutely right.

Reply:
This should be `typing.Type[OperationType]`

Reply:
Hm. Now I am confused myself.

Reply:
What I wanted to say here is that the argument is a concrete type that is a subtype of `Operation`

Reply:
So normally, `typing.Type[Operation]` should work, but `typing.Type[OperationType]` as well.
As long as you use the type only once, they should be equivalent

Reply:
This was indeed a mistake. I fixed this.